modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/properties/time/TimestampFileNameExtractor.java

```diff
@@ -20,35 +20,54 @@
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Date;
 import java.util.List;
 import java.util.TimeZone;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 import org.geotools.gce.imagemosaic.properties.PropertiesCollectorSPI;
 import org.geotools.gce.imagemosaic.properties.RegExPropertiesCollector;
+import org.geotools.util.DateRange;
+import org.geotools.util.DateTimeParser;
+import org.geotools.util.DateTimeParser.FormatAndPrecision;
 import org.geotools.util.logging.Logging;
 import org.opengis.feature.simple.SimpleFeature;
 
 /** @author Simone Giannecchini, GeoSolutions SAS */
 class TimestampFileNameExtractor extends RegExPropertiesCollector {
     private static final Logger LOGGER = Logging.getLogger(TimestampFileNameExtractor.class);
 
-    private static final TimeParser parser = new TimeParser();
+    private static final int DEFAULT_TIME_PARSER_FLAGS =
+            DateTimeParser.FLAG_GET_TIME_ON_CURRENT
+                    | DateTimeParser.FLAG_GET_TIME_ON_NOW
+                    | DateTimeParser.FLAG_GET_TIME_ON_PRESENT
+                    | DateTimeParser.FLAG_IS_LENIENT;
+
+    private static final DateTimeParser PARSER =
+            new DateTimeParser(
+                    -1, DEFAULT_TIME_PARSER_FLAGS | DateTimeParser.FLAG_SINGLE_DATE_AS_DATERANGE);
 
     private DateFormat customFormat;
 
+    private String format;
+
+    private boolean inclusive = false;
+
     public TimestampFileNameExtractor(
```

Comment:
API change here?

Reply:
mmm. actually not. that is a package private class